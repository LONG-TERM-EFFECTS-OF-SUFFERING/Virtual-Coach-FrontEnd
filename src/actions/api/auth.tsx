import axios from 'axios';
const api_url = import.meta.env.VITE_API_URL;

export const jwt_verify: any = async (token: string) => {
    const config = {
        headers: {
            'Content-Type': 'application/json'
        }
    }
    const body = JSON.stringify({ token });

    const response = await axios.post(`${api_url}/auth/jwt/verify/`, body, config);

    return response.data
}

export const jwt_create: any = async (email: string, password: string) => {
    const config = {
        headers: {
            'Content-Type': 'application/json'
        }
    };
    const body = JSON.stringify({ email, password });

    const response = await axios.post(`${api_url}/auth/jwt/create/`, body, config)

    return response.data
}

export const users_me: any = async () => {
    const jwt_token = localStorage.getItem('access')
    const config = {
        headers: {
            'Content-Type': 'application/json',
            'Authorization': `JWT ${jwt_token}`,
            'Accept': 'application/json'
        }
    }
    const response = await axios.get(`${api_url}/auth/users/me/`, config)

    return response.data
}

export const users_put: any = async (name: string) => {
    const jwt_token = localStorage.getItem('access')
    const config = {
        headers: {
            'Content-Type': 'application/json',
            'Authorization': `JWT ${jwt_token}`,
            'Accept': 'application/json'
        }
    }

    const body = JSON.stringify({ name })

    const response = await axios.put(`${api_url}/auth/users/me/`, body, config)

    return response.data
}

export const set_email: any = async (new_email: string, re_new_email: string, current_password: string) => {
    const jwt_token = localStorage.getItem('access')
    const config = {
        headers: {
            'Content-Type': 'application/json',
            'Authorization': `JWT ${jwt_token}`,
            'Accept': 'application/json'
        }
    }

    const body = JSON.stringify({ new_email, re_new_email, current_password })

    try {
        const response = await axios.post(`${api_url}/auth/users/set_email/`, body, config)
        return { data: response.data, error: false }
    } catch (err: any) {
        return { data: err.response.data, error: true }
    }
}

export const set_password: any = async (new_password: string, re_new_password: string, current_password: string) => {
    const jwt_token = localStorage.getItem('access')
    const config = {
        headers: {
            'Content-Type': 'application/json',
            'Authorization': `JWT ${jwt_token}`,
            'Accept': 'application/json'
        }
    }

    const body = JSON.stringify({ new_password, re_new_password, current_password })

    const response = await axios.post(`${api_url}/auth/users/set_password/`, body, config)

    return response.data
}