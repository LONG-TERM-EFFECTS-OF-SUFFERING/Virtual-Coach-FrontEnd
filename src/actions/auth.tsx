import axios from 'axios';


import { loadUserFail, loadUserSuccess, loginUserFail, loginUserSuccess } from '../store/slices/user/user'

const api_url = import.meta.env.VITE_API_URL;

export const login = (email: string, password: string) => async (dispatch: any) => {

    const config = {
        headers: {
            'Content-Type': 'application/json'
        }
    };
    const body = JSON.stringify({ email, password });

    await axios
        .post(`${api_url}/auth/jwt/create/`, body, config)
        .then((response) => {
            dispatch(
                loginUserSuccess(response.data)
            )
            dispatch(load_user())
        })
        .catch((err) => {
            dispatch(
                loginUserFail()
            )
        })

}

export const load_user = () => async (dispatch: any) => {
    if (localStorage.getItem('access')) {
        const config = {
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `JWT ${localStorage.getItem('access')}`,
                'Accept': 'application/json'
            }
        }
        console.log(config)
        await axios
            .get(`${api_url}/auth/users/me/`, config)
            .then((response) => {
                dispatch(
                    loadUserSuccess(response.data)
                )
            })
            .catch((err) => {
                dispatch(loadUserFail())
            })
    } else {
        dispatch(loadUserFail())
    }
}
