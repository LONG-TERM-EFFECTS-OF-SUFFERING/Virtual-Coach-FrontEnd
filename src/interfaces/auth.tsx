import { AppDispatch } from "../store/store";

export type LoginType = (email: string, password: string) => (dispatch: AppDispatch) => Promise<void>;
