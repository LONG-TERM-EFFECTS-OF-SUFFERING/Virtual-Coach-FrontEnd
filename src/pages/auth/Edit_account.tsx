import { useState } from "react"
import { PiPasswordLight, PiUserLight } from 'react-icons/pi';
import { AiOutlineEye, AiOutlineEyeInvisible } from 'react-icons/ai';

const EditAccount = () => {

  //const [formData, setFormData] = useState({
  //  name: "",
  //  password: "",
  //})

  const [showPassword, setShowPassword] = useState(false)

  const handleTogglePassword = () => {
    setShowPassword(!showPassword);
  };

  //const onChange = (e: any) => setFormData({ ...formData, [e.target.name]: e.target.value })

  //const onSubmit = async (e: any) => {
  //  e.preventDefault()
  //  console.log("SUCCESS")
  //}

  return (
    <div className="bg-gray-800 rounded-lg p-4 min-h-screen flex items-center justify-center">
      <div className="bg-gray-200 p-8 w-full md:w-96 rounded-xl">
        <h1 className="text-2xl font-bold mb-5 text-center">Editar información</h1>
        <p className="pb-4">Rellena los datos que desees modificar de tu cuenta</p>
        <form className="flex flex-col gap-4">
          <div className="relative">
            <PiUserLight className={"absolute top-2 left-2"} />
            <input type="name" className="bg-gray-100 border w-full outline-none px-4 pl-7 rounded-lg py-1" placeholder="Nombre completo" />
          </div>
          <div className="relative">
            <PiPasswordLight className="absolute top-2 left-2" />
            <input type={showPassword ? "text" : "password"} className="bg-gray-100 border w-full outline-none px-8  rounded-lg py-1" placeholder="contraseña" />
            {showPassword ? (
              <AiOutlineEye className="absolute top-2 right-2 hover:cursor-pointer" onClick={handleTogglePassword} />
            ) : (
              <AiOutlineEyeInvisible className="absolute top-2 right-2 hover:cursor-pointer" onClick={handleTogglePassword} />
            )}
          </div>
          <div>
            <button className="w-full mt-5 bg-sky-500 hover:bg-indigo-500 hover:scale-105  transition-all text-white rounded-md py-1 font-bold">
              Modificar datos
            </button>
          </div>
        </form>
      </div>
    </div>
  )
}

export default EditAccount