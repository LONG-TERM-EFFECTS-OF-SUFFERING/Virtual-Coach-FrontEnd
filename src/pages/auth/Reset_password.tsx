
const Reset_password = () => {
  return (
    <div>Reset_password</div>
  )
}

export default Reset_password