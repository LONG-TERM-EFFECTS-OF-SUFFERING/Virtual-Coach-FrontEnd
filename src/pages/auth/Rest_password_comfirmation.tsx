
const Rest_password_comfirmation = () => {
  return (
    <div>Rest_password_comfirmation</div>
  )
}

export default Rest_password_comfirmation