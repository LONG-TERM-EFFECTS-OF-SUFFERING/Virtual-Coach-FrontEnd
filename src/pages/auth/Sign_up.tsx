import React, { useState } from 'react';

const Sign_up = () => {

  const [email, setEmail] = useState('');
  const[password, setPassword] = useState('')
  const [showPassword, setShowPassword] = useState(false)

  
  

  const toggleShowPassword = () => {
    setShowPassword(!showPassword);
  };

 
  const iconOpen = <svg className="h-8 w-8 text-sky-700"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  strokeWidth="2"  strokeLinecap="round"  strokeLinejoin="round">  <path d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z" />  <circle cx="12" cy="12" r="3" /></svg>

  const iconClose = <svg className="h-8 w-8 text-sky-700"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"
  strokeWidth="2"  strokeLinecap="round"  strokeLinejoin="round">  <path d="M17.94 17.94A10.07 10.07 0 0 1 12 20c-7 
  0-11-8-11-8a18.45 18.45 0 0 1 5.06-5.94M9.9 4.24A9.12 9.12 0 0 1 12 4c7 0 11 8 11 8a18.5 18.5 0 0 1-2.16 3.19m-6.72-1.07a3 3 
  0 1 1-4.24-4.24" />  <line x1="1" y1="1" x2="23" y2="23" /></svg>

  return (
    <div className="fixed top-0 left-0 w-screen h-screen bg-opacity-50 ">

        <div className='flex h-full'>

            <div className="w-1/2 flex justify-center items-center bg-gray-800">
                
                <div className="bg-gray p-4  mx-auto my-8 text-center">
                    <p className=" text-white text-xl font-bold text-xl pb-5">¡¡De nada sirve hacer ejercicio sin tener un orden!!</p>
                    <p className="text-white text-lg">Nuestra app está diseñada para hacer tus horas de ejercicio algo más controlado</p>
                    <div className='flex justify-center pt-10'>
                        <img src="../src/logo.png" alt="Logo"></img>
                    </div>
                </div>    
            </div>

            <div className='w-1/2 flex justify-center items-center bg-sky-600'>
                <div> 
                    <div className='pb-10'>
                    <label className='text-white pb-12 font-bold'>
                        ¡¡Registrarse es gratis y toma poco tiempo!!
                    </label>
                    </div>
                <form className="bg-white shadow-md flex flex-col items-center rounded-xl px-8 pt-10 pb-8 mb-1">
                    <div className='pb-2'>
                        <label className=' font-bold'>
                            Correo electrónico
                        </label>
                    </div>  
                    <div className='pb-6'>
                        <input className=' text-2xl bg-white border-b border-sky-300 focus:border-sky-700 focus:outline-none' 
                        type="email"
                        id="email"
                        value={email}
                        onChange={(e) => setEmail(e.target.value)}>
                        </input>
                    </div>
                    <div className='pb-2'>
                        <label className=' font-bold'>
                            Contraseña
                        </label>
                    </div>  
                    <div className="relative">
                      <input
                        type={showPassword ? 'text' : 'password'}
                        id="password"
                        value={password}
                        onChange={(e) => setPassword(e.target.value)}
                        className=" text-2xl bg-white border-b border-sky-300 focus:border-sky-700 focus:outline-none"
                        />
                      <span
                        className="absolute top-1/2 right-2 transform -translate-y-1/2 cursor-pointer"
                        onClick={toggleShowPassword}
                        style={{ zIndex: 1 }}
                      >
                        {showPassword ? iconOpen : iconClose}
                      </span>
                      
                    </div>
                    <div className='pt-8'>
                        
                        <button  className ='transition-transform transform hover:scale-110 text-white bg-sky-500 hover:bg-blue-800 font-bold py-3 px-5 rounded-lg'>
                            registrarse
                        </button>
                        
                    </div>
                </form>
                <p className='text-base text-white'>¿Ya tienes una cuenta? prueba  &nbsp;     
                    <a href='/login' className='inline-block text-white hover:text-gray-300 transition-transform transform hover:scale-110'> iniciar sesión</a></p>
                </div>    
            </div>
        </div>
    </div>
  )
}

export default Sign_up