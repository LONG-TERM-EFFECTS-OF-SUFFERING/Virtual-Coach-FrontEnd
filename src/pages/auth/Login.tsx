import { useState, useEffect } from "react"
import { Link, redirect } from "react-router-dom"
import { connect } from "react-redux"
import { PiPasswordLight, PiUserLight } from 'react-icons/pi';
import { AiOutlineEye, AiOutlineEyeInvisible } from 'react-icons/ai';
import {loginType} from '../../interfaces/auth'
import { login } from '../../actions/auth';


const Login: React.FC<loginType>= ({login}) => {


  const [formData, setFormData] = useState({
    email: "",
    password: "",
  })

  const [showPassword, setShowPassword] = useState(false)

  const handleTogglePassword = () => {
    setShowPassword(!showPassword);
  };
  

  const onChange = (e:any) => setFormData({ ...formData, [e.target.name]: e.target.value });

  const { email, password } = formData;

  const onSubmit =  (e:any) => {
    e.preventDefault()
    login(email, password);
  }
  return (
    <div className="bg-white p-8 w-full md:w-96 rounded-xl">
      
      <h1 className="text-2xl font-bold mb-5 text-center">Iniciar Sesión</h1>
      <form className="flex flex-col gap-4" onSubmit={e => onSubmit(e)}>
        <div className="relative">
          <PiUserLight className={"absolute top-2 left-2"}/>
          <input type="email" className="bg-gray-100 border w-full outline-none px-4 pl-7 rounded-lg py-1"  placeholder="email" onChange={e => onChange(e)} name="email" value={email}/>
        </div>
        <div className="relative">
          <PiPasswordLight className="absolute top-2 left-2"/>
          <input type={showPassword ? "text" : "password" } className="bg-gray-100 border w-full outline-none px-8  rounded-lg py-1"  placeholder="contraseña" onChange={e => onChange(e)} name="password" value={password}/>
          {showPassword ? (
            <AiOutlineEye className="absolute top-2 right-2 hover:cursor-pointer" onClick={handleTogglePassword}/>
          ) : (
            <AiOutlineEyeInvisible className="absolute top-2 right-2 hover:cursor-pointer" onClick={handleTogglePassword}/>
          )}

        </div>
        <div>
          <button className="w-full mt-5 bg-sky-500 hover:bg-indigo-500 hover:scale-105  transition-all text-white rounded-md py-1 font-bold" type='submit'>
            Login
          </button>
        </div>
      </form>
    </div>  

  )
}

// const mapStateToProps = (state: any) => ({
//   isAuthenticated: state.auth.isAuthenticated,
// })


export default connect(null, {login})(Login)