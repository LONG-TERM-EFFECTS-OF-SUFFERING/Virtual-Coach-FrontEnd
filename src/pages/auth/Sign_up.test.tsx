import { test, expect, beforeEach, vi } from 'vitest';
import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import * as auth_api from '../../actions/api/auth'
import Sign_up from './Sign_up';

const component = (
    <Sign_up />
)

const mock_user_create = (data: any, error: boolean) => {
    return Promise.resolve({
        data,
        error
    })
}

beforeEach(() => {
    vi.resetAllMocks();
    cleanup()
})

test('Sign_up renders correctly', async () => {
    const { container } = render(component)
    expect(container).toMatchSnapshot()
})

test('Error message is shown when user_create fails', async () => {
    const data = { message: ['Error']}
    vi.spyOn(auth_api, 'user_create').mockImplementation(async () =>
        mock_user_create(data, true))

    render(component)
    const sign_up_button = screen.getByText('registrarse')
    fireEvent.click(sign_up_button)

    const alert = await screen.findByText(data.message[0])
    expect(alert).toBeDefined()
})

test('Success message is shown when user_create succeeds', async () => {
    const data = { message: ['User Created']}
    vi.spyOn(auth_api, 'user_create').mockImplementation(async () =>
        mock_user_create(data, false))

    render(component)
    const sign_up_button = screen.getByText('registrarse')
    fireEvent.click(sign_up_button)

    const alert = await screen.findByText(data.message[0])
    expect(alert).toBeDefined()
})