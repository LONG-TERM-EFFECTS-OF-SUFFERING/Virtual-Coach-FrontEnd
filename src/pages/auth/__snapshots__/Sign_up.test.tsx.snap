// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sign_up renders correctly 1`] = `
<div>
  <div
    class="fixed top-0 left-0 w-screen h-screen bg-opacity-50 "
  >
    <div
      class="flex h-full"
    >
      <div
        class="w-1/2 flex justify-center items-center bg-gray-800"
      >
        <div
          class="bg-gray p-4  mx-auto my-8 text-center"
        >
          <p
            class=" text-white text-xl font-bold text-xl pb-5"
          >
            ¡¡De nada sirve hacer ejercicio sin tener un orden!!
          </p>
          <p
            class="text-white text-lg"
          >
            Nuestra app está diseñada para hacer tus horas de ejercicio algo más controlado
          </p>
          <div
            class="flex justify-center pt-10"
          >
            <img
              alt="Logo"
              src="/src/assets/logo.png"
            />
          </div>
        </div>
      </div>
      <div
        class="w-1/2 flex justify-center items-center bg-sky-600"
      >
        <div>
          <div
            class="pb-10"
          >
            <label
              class="text-white pb-12 font-bold"
            >
              ¡¡Registrarse es gratis y toma poco tiempo!!
            </label>
          </div>
          <div
            class="mb-3"
          />
          <form
            class="bg-white shadow-md flex flex-col items-center rounded-xl px-8 pt-10 pb-8 mb-1"
          >
            <div
              class="pb-2"
            >
              <label
                class="font-bold"
              >
                 Nombre
              </label>
            </div>
            <div
              class="pb-6"
            >
              <input
                class=" text-2xl bg-white border-b border-sky-300 focus:border-sky-700 focus:outline-none"
                id="name"
                name="name"
                type="name"
                value=""
              />
            </div>
            <div
              class="pb-2"
            >
              <label
                class=" font-bold"
              >
                Correo electrónico
              </label>
            </div>
            <div
              class="pb-6"
            >
              <input
                class=" text-2xl bg-white border-b border-sky-300 focus:border-sky-700 focus:outline-none"
                id="email"
                name="email"
                type="email"
                value=""
              />
            </div>
            <div
              class="pb-2"
            >
              <label
                class=" font-bold"
              >
                Contraseña
              </label>
            </div>
            <div
              class="relative pb-6"
            >
              <input
                class=" text-2xl bg-white border-b border-sky-300 focus:border-sky-700 focus:outline-none"
                id="password"
                name="password"
                type="password"
                value=""
              />
              <span
                class="absolute top-1/2 right-2 transform -translate-y-1/2 cursor-pointer"
                style="z-index: 1;"
              >
                <svg
                  class="h-8 w-8 text-sky-700"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                >
                    
                  <path
                    d="M17.94 17.94A10.07 10.07 0 0 1 12 20c-7 
  0-11-8-11-8a18.45 18.45 0 0 1 5.06-5.94M9.9 4.24A9.12 9.12 0 0 1 12 4c7 0 11 8 11 8a18.5 18.5 0 0 1-2.16 3.19m-6.72-1.07a3 3 
  0 1 1-4.24-4.24"
                  />
                    
                  <line
                    x1="1"
                    x2="23"
                    y1="1"
                    y2="23"
                  />
                </svg>
              </span>
            </div>
            <div
              class="pb-2"
            >
              <label
                class=" font-bold"
              >
                Repite Contraseña
              </label>
            </div>
            <div
              class="relative pb-6"
            >
              <input
                class=" text-2xl bg-white border-b border-sky-300 focus:border-sky-700 focus:outline-none"
                id="re_password"
                name="re_password"
                type="password"
                value=""
              />
              <span
                class="absolute top-1/2 right-2 transform -translate-y-1/2 cursor-pointer"
                style="z-index: 1;"
              >
                <svg
                  class="h-8 w-8 text-sky-700"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                >
                    
                  <path
                    d="M17.94 17.94A10.07 10.07 0 0 1 12 20c-7 
  0-11-8-11-8a18.45 18.45 0 0 1 5.06-5.94M9.9 4.24A9.12 9.12 0 0 1 12 4c7 0 11 8 11 8a18.5 18.5 0 0 1-2.16 3.19m-6.72-1.07a3 3 
  0 1 1-4.24-4.24"
                  />
                    
                  <line
                    x1="1"
                    x2="23"
                    y1="1"
                    y2="23"
                  />
                </svg>
              </span>
            </div>
            <div
              class="pt-8"
            >
              <button
                class=" transition-transform transform hover:scale-110 text-white bg-sky-500 hover:bg-blue-800 font-bold py-3 px-5 rounded-lg"
                id="registrarse"
                type="submit"
              >
                registrarse
              </button>
            </div>
          </form>
          <p
            class="text-base text-white"
          >
            ¿Ya tienes una cuenta? prueba   
            <a
              class="inline-block text-white hover:text-gray-300 transition-transform transform hover:scale-110"
              href="/login"
            >
               iniciar sesión
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
